packages/frontend/src/components/UnderlyingData/UnderlyingDataProvider.tsx

```diff
@@ -0,0 +1,274 @@
+            state.unsavedChartVersion.metricQuery.dimensions;
+        const dimensions = explore ? getDimensions(explore) : [];
+        return dimensions.reduce((acc, dimension) => {
+            const fieldId = isField(dimension) ? getFieldId(dimension) : '';
+            if (selectedDimensions.includes(fieldId))
+                return {
+                    ...acc,
+                    [fieldId]: dimension,
+                };
+            else return acc;
+        }, {});
+    }, [explore, state.unsavedChartVersion.metricQuery.dimensions]);
+    const closeModal = useCallback(() => {
+        resetQueryResults();
+    }, [resetQueryResults]);
+
+    useEffect(() => {
+        if (
+            state.shouldFetchResults &&
+            state.unsavedChartVersion.metricQuery.dimensions.length > 0
+        ) {
+            mutate();
+            setState({ ...state, shouldFetchResults: false });
+        }
+    }, [mutate, state, setState]);
+
+    const allFields = useMemo(
+        () => (explore ? getFields(explore) : []),
+        [explore],
+    );
+    const allDimensions = useMemo(
+        () => (explore ? getDimensions(explore) : []),
+        [explore],
+    );
+
+    const viewData = useCallback(
+        (
+            value: ResultRow[0]['value'],
+            meta: TableColumn['meta'],
+            row: ResultRow,
+        ) => {
+            if (meta?.item === undefined) return;
+
+            // We include tables from all fields that appear on the SQL query (aka tables from all columns in results)
+            const rowFields = Object.keys(row);
+            const fieldsInQuery = allFields.filter((field) =>
+                rowFields.includes(getFieldId(field)),
+            );
+            const tablesInQuery = new Set([
+                ...fieldsInQuery.map((field) => field.table),
+                tableName,
+            ]);
+
+            // If we are viewing data from a metric or a table calculation, we filter using all existing dimensions in the table
+            const dimensionFilters = !isDimension(meta?.item)
+                ? Object.entries(row).reduce((acc, r) => {
+                      const [
+                          key,
+                          {
+                              value: { raw },
+                          },
+                      ] = r;
+
+                      const dimensionFilter: FilterRule = {
+                          id: uuidv4(),
+                          target: {
+                              fieldId: key,
+                          },
+                          operator: FilterOperator.EQUALS,
+                          values: [raw],
+                      };
+                      const isValidDimension = allDimensions.find(
+                          (dimension) => getFieldId(dimension) === key,
+                      );
+
+                      if (
+                          isValidDimension ||
+                          isDateDimension(allDimensions, key)
```

Comment:
I think `isValidDimension()` should return true for the date interval dimensions since it contains all the dimensions for that explore. So `isDateDimension` might be redundant

Reply:
your're right, this is no longer needed